pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -2531,6 +2531,34 @@ self: super: {
   # Restrictive upper bound on base.
   # Remove once version 1.* is released
   monad-bayes = doJailbreak super.monad-bayes;
+
+  crypt-sha512 = overrideCabal (drv: {
+    librarySystemDepends = [
+      pkgs.libxcrypt
+    ];
+    # Test failure after libxcrypt migration, reported upstrem at
+    # https://github.com/phadej/crypt-sha512/issues/13
+    doCheck = false;
+  }) super.crypt-sha512;
+
+  nano-cryptr = overrideCabal (drv: {
+    librarySystemDepends = [
+      pkgs.libxcrypt
+    ];
+  }) super.nano-cryptr;
+
+  Unixutils = overrideCabal (drv: {
+    librarySystemDepends = [
+      pkgs.libxcrypt
+    ];
+  }) super.Unixutils;
+
+  xmonad-utils = overrideCabal (drv: {
+    librarySystemDepends = [
+      pkgs.libxcrypt
+    ];
+  }) super.xmonad-utils;
+
```

Comment:
Haskell changes look good in principle. (Can’t comment on the overall migration.)

Well, technically the `librarySystemDepends` override belong into `configuration-nix.nix` because they are permanent and will affect every version of those packages.

Only the `dontCheck` override belongs into `configuration-common.nix` because this is a temporary fix and will hopefully not be required by future versions of this package.

Reply:
Feel free to move these around, ideally by pushing fixup commits.

Reply:
I think we can save ourselves quite a bit of work by adding a rule to `cabal2nix` that causes it to pick `libxcrypt` if it sees:

```cabal
   extra-libraries:  crypt
```

This could be as easy as changing [this line](https://github.com/NixOS/cabal2nix/blob/3a2abb2e24afe70da2e384f27ebc358ee9e0ebd3/src/Distribution/Nixpkgs/Haskell/FromCabal/Name.hs#L39).

Reply:
@mweinelt What happens if we start injecting `libxcrypt` into Haskell builds on our regular branch already? Would glibc take precedence? Would it conflict?

Reply:
Honestly no idea.

Reply:
I don't think it would conflict.  `ldd path/to/binary` should tell you if libxcrypt is used.